src/media/js/addon/actions/addon.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+import {createAction} from 'redux-actions';
+import req from 'request';
+import Url from 'urlgray';
+import urlJoin from 'url-join';
+
+
+export const FETCH_OK = 'ADDON_ADDON__FETCH_OK';
+const fetchOk = createAction(FETCH_OK);
+
+export const FETCH_VERSIONS_OK = 'ADDON_ADDON__FETCH_VERSIONS_OK';
+const fetchVersionsOk = createAction(FETCH_VERSIONS_OK);
+
+
+export function fetch(addonSlug) {
+  /*
+    Fetch add-on.
+  */
+  return (dispatch, getState) => {
+    const apiArgs = getState().apiArgs || {};
+    const addonUrl = Url(
+      urlJoin(process.env.MKT_API_ROOT, 'extensions/extension/', addonSlug)
+    ).q(apiArgs);
+
+    if (process.env.NODE_ENV === 'test') {
```

Comment:
Can you mock it at the network level in test code? Seems weird to me to see this here.

I'm not sure of how `dispatch` works but it seems like this is asynchronous with the production code and synchronous with the test code.